File: nodes/delay/delay.process.js
export default async ({inputs, settings, config}) => {

  // Get delay time with safety limits
  const requestedDelay = Math.max(0, Number(inputs.delay_ms) || 0);
  const maxDelay = Math.max(0, Number(inputs.max_delay_ms) || 60000);
  const actualDelay = Math.min(requestedDelay, maxDelay);
  
  // Create a promise that resolves after the delay
  await new Promise(resolve => setTimeout(resolve, actualDelay));
  
  // Return the original value and the actual delay time
  return {
    result: inputs.value,
    actual_delay_ms: actualDelay
  };
}; 

File: nodes/delay/delay.process.py
import asyncio

async def process(inputs, settings, config, nodeConfig):
  # Get delay time with safety limits
  requested_delay = max(0, float(inputs.get("delay_ms", 0)))
  max_delay = max(0, float(inputs.get("max_delay_ms", 60000)))
  actual_delay = min(requested_delay, max_delay)
  
  # Wait for the specified time
  await asyncio.sleep(actual_delay / 1000)  # Convert ms to seconds
  
  # Return the original value and the actual delay time
  return {
    "result": inputs.get("value"),
    "actual_delay_ms": actual_delay
  } 